src/Tokenizer/Analyzer/SwitchAnalyzer.php

```diff
@@ -94,30 +82,59 @@ private function getCasesEnd(Tokens $tokens, $casesStartIndex)
             return $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $casesStartIndex);
         }
 
-        $depth = 1;
         $index = $casesStartIndex;
-        while ($depth > 0) {
-            /** @var int $index */
-            $index = $tokens->getNextMeaningfulToken($index);
+        while ($index < $tokens->count()) {
+            $index = $this->getNextSameLevelToken($tokens, $index);
 
             if ($tokens[$index]->isGivenKind(T_ENDSWITCH)) {
-                --$depth;
-
-                continue;
+                break;
             }
+        }
 
-            if (!$tokens[$index]->isGivenKind(T_SWITCH)) {
-                continue;
-            }
-            $index = $this->getCasesStart($tokens, $index);
-            if ($tokens[$index]->equals(':')) {
-                ++$depth;
+        $afterEndswitchIndex = $tokens->getNextMeaningfulToken($index);
+
+        $afterEndswitchToken = $tokens[$afterEndswitchIndex];
+
+        return $afterEndswitchToken->equals(';') ? $afterEndswitchIndex : $index;
```

Comment:
I think `->equalsAny([';', [T_CLOSE_TAG]])`
to support

```php
<?php switch ($foo) : case 10;
                switch ($bar) : case "a": return "b"; case "c"; return "d"; case "e": return "f"; endswitch;
                return;
                case 100: return false; endswitch ?>  <?php echo 1;
```

Reply:
This is already supported - if after switch token we have closing tag then the end of switch is the switch token, not the closing tag. I've added a test case to show it, see it, and the one above to see the difference in switch end.

Reply:
Thanks for the test!
Maybe I just don't understand the meaning of case end here well enough.
So
```php
<?php
switch (time()):
    case 100: return false;
endswitch ;
```
gives end index 22, which is ';'

```php
<?php
switch (time()):
    case 100: return false;
endswitch ?> <?php echo 1;
```
gives end 20 which is `endswitch`

why aren't both `endswitch` here, or the first `;` and the second other `?>` ?

Reply:
The reasoning is: take the next statement and the previous token before it is the end of the current (the `switch`) one.

So for `switch ($x) { /* ... */ } nextStatement();` the end is `}`.
For `switch ($x): /* ... */ endswitch; nextStatement();` the end is `;` (after `endswitch`).
For `switch ($x): /* ... */ } endswitch ?>  <?php nextStatement();` the end is `endswitch` as `?>` is next statement - or is it not?

Reply:
I see, yeah, so from my POV I think for `switch ($x): /* ... */ } endswitch ?>` the close tag `?>` is what ends the `switch` , without it the statement is complete and invalid, I think `;` and `?>` have the same meaning here

Reply:
That makes sense - I've updated the tests and code.

Reply:
thanks!